dockstore-integration-testing/src/test/java/io/dockstore/webservice/UserResourceOpenApiIT.java

```diff
@@ -421,4 +429,106 @@ void testAdminAndCuratorFields() throws ApiException {
+        PrivilegeRequest robotPrivileges = new PrivilegeRequest();
+        robotPrivileges.setMetricsRobot(true);
+        adminApi.setUserPrivileges(robotPrivileges, robotId);
+
+        // NOT able to add more privileges to an existing metrics robot.
+        assertThrowsCode(HttpStatus.SC_BAD_REQUEST, () -> adminApi.setUserPrivileges(robotPlusAdminPrivileges, robotId));
+
+        // NOT able to remove metrics robot privileges.
+        PrivilegeRequest noPrivileges = new PrivilegeRequest();
+        assertThrowsCode(HttpStatus.SC_BAD_REQUEST, () -> adminApi.setUserPrivileges(noPrivileges, robotId));
+
+        // Confirm that the metrics robot does not have any other privileges.
+        User robot = getUser(adminApi, robotUsername);
+        assertTrue(robot.isMetricsRobot());
+        assertFalse(robot.isIsAdmin());
+        assertFalse(robot.isCurator());
+        assertNull(robot.getPlatformPartner());
+
+        // The robot user should be able to access the metrics submission endpoints, and should NOT be able to access other authenticated endpoints.
+        // We don't synthesize a valid metrics request here, but instead check the status code to determine "how far" the request got.
+        assertThrowsCode(HttpStatus.SC_UNPROCESSABLE_ENTITY, () -> new ExtendedGa4GhApi(getOpenAPIWebClient(robotUsername, testingPostgres)).executionMetricsPost(new ExecutionsRequestBody(), Partner.TERRA.name(), "malformedId", "malformedVersionId", null));
+        assertThrowsCode(HttpStatus.SC_FORBIDDEN, () -> robotApi.changeUsername("newname"));
+
+        // Update token ID sequence number so that it doesn't collide with existing tokens.
+        testingPostgres.runUpdateStatement("alter sequence token_id_seq increment by 50 restart with 100");
+
+        // Only admins should be able to create metrics robot tokens, and only for a metrics robot user.
+        // All other combinations of initiator and target users should fail.
+        for (UsersApi initiator: List.of(anonApi, userApi, adminApi, robotApi)) {
+            for (String target: List.of(userUsername, adminUsername, robotUsername)) {
+                if (!(initiator == adminApi && target == robotUsername)) {
+                    int expectedCode;
+                    if (initiator == anonApi) {
+                        expectedCode = HttpStatus.SC_UNAUTHORIZED;
+                    } else if (initiator == adminApi) {
+                        expectedCode = HttpStatus.SC_BAD_REQUEST;
+                    } else {
+                        expectedCode = HttpStatus.SC_FORBIDDEN;
+                    }
+                    User targetUser = getUser(adminApi, target);
+                    assertThrowsCode(expectedCode, () -> initiator.createMetricsRobotToken(targetUser.getId()));
+                }
+            }
+        }
+
+        // Admin should be able to delete a robot Dockstore token.
+        TokenUser token = getDockstoreToken(adminApi, robotId);
```

Comment:
I don't think there's anything special about it, but would be nice to demonstrate that a robot user cannot delete their own token (and thus mess up other robots)

Reply:
Good idea, added.